FastSurferCNN/data_loader/conform.py

```diff
@@ -560,33 +661,57 @@ def conform(
     new_img = nib.MGHImage(sctype(mapped_data), affine, h1)
 
     # make sure we store uchar
+    from nibabel.freesurfer import mghformat
     try:
         new_img.set_data_dtype(target_dtype)
-    except nib.freesurfer.mghformat.MGHError as e:
-        if "not recognized" in e.args[0]:
-            codes = set(
-                k.name
-                for k in nib.freesurfer.mghformat.data_type_codes.code.keys()
-                if isinstance(k, np.dtype)
-            )
-            print(
-                f'The data type "{options.dtype}" is not recognized for MGH images, switching '
-                f'to "{new_img.get_data_dtype()}" (supported: {tuple(codes)}).'
-            )
+    except mghformat.MGHError as e:
+        if "not recognized" not in e.args[0]:
+            raise
+        dtype_codes = mghformat.data_type_codes.code.keys()
+        codes = set(k.name for k in dtype_codes if isinstance(k, np.dtype))
+        print(
+            f"The data type '{options.dtype}' is not recognized for MGH images, "
+            f"switching to '{new_img.get_data_dtype()}' (supported: {tuple(codes)})."
+        )
 
     return new_img
 
 
+def is_resampling_vox2vox(
+        vox2vox: npt.NDArray[float],
+        eps: float = 1e-6,
+) -> bool:
+    """
+    Check whether the affine is resampling or just reordering.
+
+    Parameters
+    ----------
+    vox2vox : np.ndarray
+        The affine matrix.
+    eps : float, default=1e-6
+        The epsilon for the affine check.
+
+    Returns
+    -------
+    bool
+        The result.
+    """
+    _v2v = np.abs(vox2vox[:3, :3])
+    # check 1: have exactly 3 times 1/-1 rest 0, check 2: all 1/-1 or 0
+    return abs(_v2v.sum() - 3) > eps or np.any(np.maximum(_v2v, abs(_v2v - 1)) > eps)
```

Comment:
shouldn't this be abs(_v2v.sum()) - 3 ? and also abs(_v3v) - 1 ?

Reply:
v2v is already abs in line 699